tests/integration/cli/detection/test_detection.py

```diff
@@ -35,6 +39,12 @@
     "train_params": ["params", "--learning_parameters.num_iters", "1", "--learning_parameters.batch_size", "4"],
 }
 
+num_iters_per_model = {
```

Comment:
Should we have below long interactions with integration test?

Reply:
For the same point, we need to reduce the elapsed time for integration tests.

IMO, we just need to check functionality in the integration tests executed at every PR.
So, how about reducing `num_iters`?

Reply:
Actually, there is a functionality in XAI, which suppose to generate saliency maps only for predictions. I need a trained model to test it. So if the model is not trained - no confident predictions (no threshold is passed).

As a solution, I moved this check into the e2e, keeping 1 iter for `tests/integration/cli/detection/test_detection.py` Does it work for you?